src/parser/ParsedQuery.h

```diff
@@ -357,9 +359,9 @@ class ParsedQuery {
     // Select All (Select '*')
     // or
     // explicit variables selection (Select 'var_1' ... 'var_n')
-    [[nodiscard]] const auto& getSelectedVariable() const {
-      return isAsterisk() ? _variablesFromQueryBody
-                          : std::get<std::vector<string>>(_varsOrAsterisk);
+    [[nodiscard]] const auto& getSelectedVariables() const {
+      return isAsterisk() ? orderedVariablesFromQueryBody()
+                          : getManuallySelectedVariables();
```

Comment:
This class is now much much cleaner, thank you.
Two things that I missed the last time:
Is the `orderedVariablesFromQueryBod()` function still needed outside of this class? Otherwise you can completely remove it and return the `_variablesFromQueryBody` directly in the `getSelectedVariables()` function.

Reply:
The typdef `SelectVarsOrAsterisk` is visible outside of this class, although it is not needed, and it has a confusing name.
Since you use it exactly one, you can remove it and make the `_varsOrAsterisk` directly a `std::variant<...`>

3.
If the `getManuallySelectedVariables()` function is still needed outside this class, then it should have an `AD_CHECK(isVariables);` to get a good error message. But if it's not needed outside, it should be private.


`isVariables` -> `isManuallySelectedVariables`
`isAsterisk` -> `isAllVariablesSelected`
(Then they match the `set...`function.